ipaserver/install/ipa_idrange_fix.py

```diff
@@ -0,0 +1,1073 @@
+"""Tool to analyze and fix IPA ID ranges"""
+# Authors: Aleksandr Sharov <asharov@redhat.com>
+#
+# Copyright (C) 2024  Red Hat
+# see file 'COPYING' for use and warranty information
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import ldap
+
+from ipalib import api, errors
+from ipapython.admintool import AdminTool
+from ipapython.dn import DN
+from ipapython import ipautil
+
+logger = logging.getLogger(__name__)
+
+
+class IDRange:
+    """Class for ID range entity"""
+
+    def __init__(self, **kwargs):
+        self.last_id: int = None
+        self.last_base_rid: int = None
+        self.last_secondary_rid: int = None
+        self.name: str = kwargs.get('name')
+        self.size: int = kwargs.get('size')
+        self.first_id: int = kwargs.get('first_id')
+        self.base_rid: int = kwargs.get('base_rid')
+        self.secondary_base_rid: int = kwargs.get('secondary_base_rid')
+        self.type: str = kwargs.get('type')
+        self.suffix: str = kwargs.get('suffix')
+        self.dn: str = kwargs.get('dn')
+        if self.first_id is not None and self.size is not None:
+            self.count()
+
+    def count(self) -> None:
```

Comment:
I'm conflicted on this. I think at a minimum it should be a weak private (_count) but why can't think be part of __init__ and have this generation as part of the constructor? I also wonder about **kwargs since AFAICT no arguments are ever passed into it. Now maybe it's clearer to get an empty object and then populate it, like in the "for entry in ranges" loop does. Passing in these dozen or so arguments would probably be less readable. Like I said, I'm conflicted and trying not to bikeshed. I'd at least get rid of the unused kwargs.

Reply:
I've intended to both use kwargs and empty init, and used them at some point, but indeed found it much more readable to use an empty instance that I fill in. Removed kwargs, renamed method to _count(), now there's no reason to call it in init.